modules/nf-core/kallistobustools/count/main.nf

```diff
@@ -4,8 +4,8 @@ process KALLISTOBUSTOOLS_COUNT {
 
     conda "${moduleDir}/environment.yml"
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'https://depot.galaxyproject.org/singularity/kb-python:0.27.2--pyhdfd78af_0' :
-        'biocontainers/kb-python:0.27.2--pyhdfd78af_0' }"
```

Comment:
Once everything works, please update the modules in `nf-core/modules` and re-install them into the pipeline using `nf-core modules update`. We don't allow modules in the pipelines to diverge from the central version stored in `modules`.

Reply:
Thank you, I have put in a request. 
I appreciate the clarification regarding the `nf-core/modules` source – so I would need to make changes in https://github.com/nf-core/modules/tree/master/modules/nf-core/kallistobustools, then pull them in.

If I understand correctly, these module essentially have the input/output conventions and a link to the docker image, which does not actually work. 
Is there any way to rebuild the binary for the kallisto dependency?

Reply:
Exactly. Whenever possible, we try to use the docker images provided by [Biocontainers](https://biocontainers.pro/), which are derived from the [bioconda recipe](https://github.com/bioconda/bioconda-recipes/blob/master/recipes/kb-python/meta.yaml). So if the container is broken, it would be ideal to make it work there. If this is not possible for whatever reason, we can also use custom docker images in `nf-core/modules`. 

To just rebuild, you can make a PR to bioconda recipes that increases the build number: https://github.com/bioconda/bioconda-recipes/blob/2094e1c611eedfe500a145bd442640f0440f68a5/recipes/kb-python/meta.yaml#L14

Reply:
I went into a Docker session and checked the container manually. It runs everything fine: `kb` successfully completes `ref` and `count` on small test datasets using any container. Yet only pre-0.28.0 containers work for `nf-core/scrnaseq`. Any ideas?

Reply:
The bioconda recipe has been fixed. I have opened a PR at https://github.com/nf-core/modules/pull/4981; this should be complete relatively soon. In the meantime, I have updated the tests and adjusted the downstream mtx operations.

Reply:
@grst  the nf-core module PR is merged and I have updated the kb modules in the PR branch. It still fails linting due to other files – not sure what to do; would love any suggestions. Running `nf-core lint` locally does not give any failures.